### ISSUE TITLE
Add setting to manually set Device ID please

### ISSUE THREAD

**Issue by ale5000-git** [microg/android_packages_apps_GmsCore#521]
Add setting to manually set Device ID please.

The one that with GApps you can get using:
`adb shell 'sqlite3 /data/data/com.google.android.gsf/databases/gservices.db "select * from main where name = \"android_id\";"`

Ref: https://www.xda-developers.com/how-to-fix-device-not-certified-by-google-error/

**Comment by Nanolx** [comment]
microG's DroidGuardHelper is outdated since the latest update from DroidGuard Google has pushed, so registering the ID will not fix any issue with microG.

You can test this by using the 'Device ID' app which will show the spoofed GSF-ID by microG.

Though, of course being able to choose the spoofed ID would make it possible to only register one to Google, but what happens, when two users spoof the same ID? Or when two devices with the same ID are simultaneously active?

**Comment by ale5000-git** [comment]
I know, I'm just asking a thing that will be useful later.
It would be mainly to reuse existing GApps Device ID but an experienced person can use it also in other ways.

**Comment by lazerl0rd** [comment]
@Nanolx is there any other way to pass safteynet with microG at the moment?

**Comment by ale5000-git** [comment]
@lazerl0rd: This is not directly related to SafetyNet, lazerl0rd  may not work also if the device pass certification.
The SafetyNet problem is separate.

**Comment by Nanolx** [comment]
@ale5000-git Yes, I was just wondering what would happen IF that's the case, so it should be considered an "experimental" feature.

@lazerl0rd the SafetyNet issue is regarding an update to DroidGuard Google has pushed last week. microG's DroidGuard Helper needs to be updated accordingly.

There's currently no way to pass CTS or even BasicIntegrity with microG **even if you register the (spoofed) Device ID**, thus Google Apps will see the device as uncertified and stop working (GMail and stuff), including all other apps that check the SN attestation (Nintendo games and stuff).

Though all that's already discussed over at issue #510 

possible workarounds:

* use a second device without private data with GApps for your gaming fix
* wait until our savior @mar-v-in descends and blesses us with some shiny new code that shall allow us to safely enter the gates of mordor

**Comment by lazerl0rd** [comment]
Well @ale5000-git for the Device ID use the app mentioned on xda to get it (worked for me and now I can download form Play Store at least while using microG (I've even got a few Google Apps to work))

@Nanolx thanks mate

**Comment by ucecus** [comment]
same case...i cant change android id in google device registration menu using xpoed module device emulator..
